### ISSUE TITLE
Instruction on how to set up in Nuxt project

### ISSUE THREAD

**Issue by louis-l** [Nickersoft/push.js#266]
Hi there,

Thanks for the awesome library. I started to look around and figure out if this is suitable for my existing Nuxt project (using PWA module).

I want to push notifications to desktop and mobile devices no matter the web app is being opened (active) or running in background (using service worker).
I have read the https://pushjs.org/docs/options but still not sure how to. Also I cannot test this in local environment, so the only way for me to try and test this is to set up a staging environment, which requires a bit of work. Therefore, it would be great if you can help me to answer these questions:

- do I need to manually create **serviceWorker.js** file and copy the content from https://github.com/Nickersoft/push.js/blob/master/src/serviceWorker.js
- do you have examples (or can you point me to the correct direction) of how to show notifications in service worker (so that it works in background mode)?
- I know this is not relevant, but can you advice what is the best practice to use this library to show notifications to a certain group of users? For example: when new order is placed, I want to push notification to users that have access to that new order only

Thank you.

**Comment by theLufenk** [comment]
Hi @louis-l 
Thanks for trying out the library.

It appears to me that what you intend to do is send targeted notifications to your users, possibly triggered by a server.

This library, admittedly named ambiguously, does not implement the PUSH api. Only the NOTIFICATION api.

Collecting subscribers and sending targeted push notifications would entail using a service such as Firebase.

Here is an awesome and interesting guide by google developers: https://developers.google.com/web/ilt/pwa/introduction-to-push-notifications#how_web_push_works

This should get you on the path I reckon.

Also as a suggestion, I have had pretty pleasant experience with One Signal as a platform for push notifications. You can check that out here: https://onesignal.com/


Hope it helps :)

**Comment by louis-l** [comment]
Thanks @theLufenk, really appriciate your helps here.

I came across OneSignal before, even through they have segments to send notifications to different users, but it requires a bit of work to sync OneSignal subscribers with our internal users in order to correctly push notifications based on our internal user IDs.

Anyway, I found https://github.com/laravel-notification-channels/webpush which seems to be the solution that Im looking for. Will try it out.

Thanks again 👍
